packages/visual-editor/src/hooks/useBreakpoints.ts

```diff
@@ -50,7 +50,7 @@ export const useBreakpoints = (breakpoints: Breakpoint[]) => {
 
   const activeBreakpointIndex = getActiveBreakpointIndex(
     breakpoints,
-    mediaQueryMatches,
+    initialMediaQueryMatches || mediaQueryMatches,
```

Comment:
Mmmmh, this looks a bit shady to be honest.

This would mean that we never consume `mediaQueryMatches` which is the variable that gets updated when resizing the canvas. So this change would mean that we only show the designs of the initial canvas size and don't care anymore when you switch between desktop/ table/ mobile view.

If we really wanted to remove this feature, we should rather drop the whole state variable and effect instead of having this shady overwrite.

Reply:
We should probably move this deterministic logic into the `getActiveBreakPointIndex` method that comes from core so the logic can be reused by the other breakpoint functionality we have in various places.

Reply:
We have the exact same logic already in the SDK in the similar called file `useBreakpoints` [here](https://github.com/contentful/experience-builder/blob/6c7b72264e91029b78e349e37cf6195b2dc4b9eb/packages/experience-builder-sdk/src/hooks/useBreakpoints.ts#L24). So yeah, sounds like something to pull into core. Though, I'm wondering a bi right now why we didn't go with native media queries in delivery mode - the current approach is a very JS-heavy one which might not work always for the customer as expected I think. Not sure if this would affect SSG applications in any way actually.